tests/test_restrictions.py

```diff
@@ -0,0 +1,143 @@
+import pandas as pd
+from pandas.util.testing import assert_series_equal
+
+from zipline.finance.restrictions import (
+    RESTRICTION_STATES,
+    Restriction,
+    InMemoryRestrictions,
+    StaticRestrictions,
+    NoopRestrictions,
+)
+from zipline.testing.fixtures import (
+    WithDataPortal,
+    ZiplineTestCase,
+)
+
+str_to_ts = lambda dt_str: pd.Timestamp(dt_str, tz='UTC')
+
+
+class RestrictionsTestCase(WithDataPortal, ZiplineTestCase):
+
+    ASSET_FINDER_EQUITY_SIDS = 1, 2, 3
+
+    @classmethod
+    def init_class_fixtures(cls):
+        super(RestrictionsTestCase, cls).init_class_fixtures()
+        cls.ASSET1 = cls.asset_finder.retrieve_asset(1)
+        cls.ASSET2 = cls.asset_finder.retrieve_asset(2)
+        cls.ASSET3 = cls.asset_finder.retrieve_asset(3)
+
+    def test_in_memory_restrictions(self):
+
+        restrictions = [
+            Restriction(
+                self.ASSET1,
+                str_to_ts('2011-01-06'),
+                RESTRICTION_STATES.FROZEN
+            ),
+            Restriction(
+                self.ASSET1,
+                str_to_ts('2011-01-04'),
+                RESTRICTION_STATES.FROZEN
+            ),
+            Restriction(
+                self.ASSET1,
+                str_to_ts('2011-01-05'),
+                RESTRICTION_STATES.ALLOWED
+            ),
+            Restriction(
+                self.ASSET2,
+                str_to_ts('2011-01-03'),
+                RESTRICTION_STATES.FROZEN
+            ),
+        ]
+
+        rlm = InMemoryRestrictions(restrictions)
+
+        # Check the restrictions are ordered by dt for each sid
+        self.assertEqual(
```

Comment:
I don't think we should be making assertions about the internal representation of the data that's chosen here.  As long as the `Restrictions` class behaves as expected, it should be free to store the data however it wants.